### PR TITLE
Remove workaround for session locking bug

### LINKED ISSUE #1596: Remove RLock from nidcpower
In #1594 we created fancy functions for `lock` and `unlock` that use a class global [RLock](https://docs.python.org/3.6/library/threading.html#rlock-objects) instead of the driver lock.  A consequence of this change is all NI-DCPower sessions are required to obtain the lock before they can make a driver call.  As a result, the lock is no longer a session lock but a driver lock and the docstrings for `lock` and `unlock` are misleading.

Once NI-DCPower supports `lock` and `unlock` for independent channels sessions, we should remove the fancy functions and return to using the driver methods. 

The affected methods are marked with todo tags in `session.py::_SessionBase`:
- `session.py::_SessionBase::lock`
- `session.py::_SessionBase::unlock`.

To remove the RLock, perform the following:
1. Delete:
    - `src/nidcpower/templates/session.py/fancy_lock.py.mako`
    - `src/nidcpower/templates/session.py/fancy_unlock.py.mako`
2. Change the metadata to use the lock and unlock templates defined in `build/templates/session.py`
    - Change `'session_filename': 'fancy_lock'` to `'session_filename': 'lock'` in `LockSession` of [functions.py](https://github.com/ni/nimi-python/blob/c631e3d8d034ac8c2ec8224363b78d5fd67b82af/src/nidcpower/metadata/functions.py#L1962).
    - Change `'session_filename': 'fancy_unlock'` to `'session_filename': 'unlock'` in `UnlockSession` of [functions.py](https://github.com/ni/nimi-python/blob/c631e3d8d034ac8c2ec8224363b78d5fd67b82af/src/nidcpower/metadata/functions.py#L2818).

### PR DESCRIPTION
**Pull request by dylan-matthew-turner**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [x] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.
- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

Multi-instrument sessions in NI-DCPower runtime did not support `niDCPower_LockSession` and
`niDCPower_UnlockSession` until version 21.5. For this, a workaround was introduced in the Python module. This has been fixed in the driver runtime and thus the workaround is being removed.

- Adds test failure for using RLock
- Removes src/nidcpower/templates/session.py/fancy_lock.py.mako and src/nidcpower/templates/session.py/fancy_unlock.py.mako
- Modifies functions.py to use lock/unlock instead of removed fancy_lock/fancy_unlock

### List issues fixed by this Pull Request below, if any.

- Fix #1596 

### What testing has been done?

- Test targeting data stored to use RLock fails in master but succeeds after changes applied
- All previous system tests pass

### CODE REVIEW

**Comment by marcoskirsch** [comment]
@dylan-matthew-turner the title and description paraphrase the diff. Please modify to explain why the change is being made and the high level purpose, which adds value.

Example: "Remove workaround for niDCPower_LockSession / niDCPower_UnlockSession"...

**Review by marcoskirsch — CHANGES_REQUESTED** [review]
PLease fix title and description

File: CHANGELOG.md
Comment (sbethur) [inline CHANGELOG.md:39]: Changelog should be used only for customer-visible changes.

The use of RLock had a minor enough customer impact (Access to multiple nidcpower sessions are serialized) that we decided not to mention in the Changelog. So removal of it doesn't need to be mentioned either.

File: src/nidcpower/system_tests/test_system_nidcpower.py
Comment (sbethur) [inline src/nidcpower/system_tests/test_system_nidcpower.py:729]: This is a good test to run as part of "dev testing" but don't think it makes a good system test to be used for regression. Outside of this PR, the test doesn't provide sufficient context or utility for maintainers. I'd say remove it.

**Review by sbethur — CHANGES_REQUESTED** [review]

**Review by marcoskirsch — APPROVED** [review]
Waiting on changes based on Shreyas' feedback.

**Review by sbethur — APPROVED** [review]

**Pull request merged**
